{"version":3,"sources":["otherwise.js"],"names":["_curry2","require","_assertPromise","otherwise","f","p","then","module","exports"],"mappings":"AAAA,IAAIA,OAAO,GAAgBC,OAAO,sBAAlC;;AAEA,IAAIC,cAAc,GAAgBD,OAAO,6BAAzC;;AA+BA,IAAIE,SAAS,GAAgBH,OAAO,CAAC,SAASG,SAAT,CAAmBC,CAAnB,EAAsBC,CAAtB,EAAyB;AAC5DH,EAAAA,cAAc,CAAC,WAAD,EAAcG,CAAd,CAAd;;AAEA,SAAOA,CAAC,CAACC,IAAF,CAAO,IAAP,EAAaF,CAAb,CAAP;AACD,CAJmC,CAApC;;AAKAG,MAAM,CAACC,OAAP,GAAiBL,SAAjB","sourcesContent":["var _curry2 = /*#__PURE__*/require('./internal/_curry2');\n\nvar _assertPromise = /*#__PURE__*/require('./internal/_assertPromise');\n\n/**\n * Returns the result of applying the onFailure function to the value inside\n * a failed promise. This is useful for handling rejected promises\n * inside function compositions.\n *\n * @func\n * @memberOf R\n * @category Function\n * @sig (e -> b) -> (Promise e a) -> (Promise e b)\n * @sig (e -> (Promise f b)) -> (Promise e a) -> (Promise f b)\n * @param {Function} onFailure The function to apply. Can return a value or a promise of a value.\n * @param {Promise} p\n * @return {Promise} The result of calling `p.then(null, onFailure)`\n * @see R.then\n * @example\n *\n *      var failedFetch = (id) => Promise.reject('bad ID');\n *      var useDefault = () => ({ firstName: 'Bob', lastName: 'Loblaw' })\n *\n *      //recoverFromFailure :: String -> Promise ({firstName, lastName})\n *      var recoverFromFailure = R.pipe(\n *        failedFetch,\n *        R.otherwise(useDefault),\n *        R.then(R.pick(['firstName', 'lastName'])),\n *      );\n *      recoverFromFailure(12345).then(console.log)\n */\n\n\nvar otherwise = /*#__PURE__*/_curry2(function otherwise(f, p) {\n  _assertPromise('otherwise', p);\n\n  return p.then(null, f);\n});\nmodule.exports = otherwise;"]}